ush/python/pygfs/task/stat_analysis.py

```diff
@@ -0,0 +1,185 @@
+                    add_to_datetime, to_timedelta,
+                    Task,
+                    parse_j2yaml,
+                    logit)
+from pygfs.jedi import Jedi
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class StatAnalysis(Task):
+    """
+    Class for JEDI-based global stat analysis tasks
+    """
+    @logit(logger, name="StatAnalysis")
+    def __init__(self, config: Dict[str, Any]):
+        """
+        Constructor global stat analysis task
+        This method will construct a global stat analysis task.
+        This includes:
+        - extending the task_config attribute AttrDict to include parameters required for this task
+        - instantiate the Jedi attribute objects
+        Parameters
+        ----------
+        config: Dict
+            dictionary object containing task configuration
+        Returns
+        ----------
+        None
+        """
+        super().__init__(config)
+
+        _res = int(self.task_config.CASE[1:])
+        _window_begin = add_to_datetime(self.task_config.current_cycle, -to_timedelta(f"{self.task_config.assim_freq}H") / 2)
+
+        # Create a local dictionary that is repeatedly used across this class
+        local_dict = AttrDict(
+            {
+                'npx_ges': _res + 1,
+                'npy_ges': _res + 1,
+                'npz_ges': self.task_config.LEVS - 1,
+                'npz': self.task_config.LEVS - 1,
+                'npz_anl': self.task_config.LEVS - 1,
+                'STAT_WINDOW_BEGIN': _window_begin,
+                'STAT_WINDOW_LENGTH': f"PT{self.task_config.assim_freq}H",
+                'OPREFIX': f"{self.task_config.RUN}.t{self.task_config.cyc:02d}z.",
+                'APREFIX': f"{self.task_config.RUN}.t{self.task_config.cyc:02d}z.",
+                'GPREFIX': f"gdas.t{self.task_config.previous_cycle.hour:02d}z."
+            }
+        )
+        # Extend task_config with local_dict
+        self.task_config = AttrDict(**self.task_config, **local_dict)
+
+    @logit(logger)
+    def initialize(self) -> None:
+        """
+        Initialize a global stat analysis
+        This method will initialize a global stat analysis.
+        This includes:
+        - initialize JEDI applications
+        - copying stat files
+        Parameters
+        ----------
+        None
+        Returns
+        ----------
+        None
+        """
+        # Create dictionary of Jedi objects
+        # Expected keys are what must be included from the JEDI config file. We can
+        # then loop through ob space list from scripts/exglobal_analysis_stats.py
+        expected_keys = ['aero', 'snow']
+        self.jedi_dict = Jedi.get_jedi_dict(self.task_config.JEDI_CONFIG_YAML, self.task_config, expected_keys)
+
+        logger.info(f"Copying files to {self.task_config.DATA}/stats")
+
+        # Loop through ob space list
+        for OB in self.task_config.STAT_OBS:
+            logger.info(f"Working on current observation: {OB}")
```

Comment:
this is snow and aero right? then it's not an 'observation'

Reply:
What would be a good replacement for this?

Reply:
yes I'd suggest something like `f"Working on current component: {anl}"`